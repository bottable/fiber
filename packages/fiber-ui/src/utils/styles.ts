import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
