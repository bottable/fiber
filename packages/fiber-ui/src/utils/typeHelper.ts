export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
