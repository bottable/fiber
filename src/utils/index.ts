export * from './typeHelper'
export * from './mergeProps'
