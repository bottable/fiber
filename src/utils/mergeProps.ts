import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
