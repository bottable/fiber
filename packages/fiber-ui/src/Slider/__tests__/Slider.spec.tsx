import { Slider } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Slider', () => {
  it('renders correctly', () => {
    const slider = render(
      <>
        <div>
          <Slider defaultValue={30} />
          <br />
          <Slider value={30} />
        </div>
        <div>
          <Slider defaultValue={30} step={2} />
          <br />
          <Slider defaultValue={30} step={10} />
        </div>
        <div>
          <Slider defaultValue={30} disabled />
        </div>
        <div>
          <div style={{ height: 300 }}>
            <Slider defaultValue={30} vertical />
          </div>
        </div>
        <div>
          <Slider
            defaultValue={30}
            marks={{
              0: '0°C',
              26: '26°C',
              37: '37°C',
              100: {
                style: {
                  color: '#f50'
                },
                label: <strong>100°C</strong>
              }
            }}
          />
          <br />
          <div style={{ height: 300 }}>
            <Slider
              defaultValue={30}
              vertical
              marks={{
                0: '0°C',
                26: '26°C',
                37: '37°C',
                100: {
                  style: {
                    color: '#f50'
                  },
                  label: <strong>100°C</strong>
                }
              }}
            />
          </div>
        </div>
      </>
    )

    expect(slider).toMatchSnapshot()
  })
})
