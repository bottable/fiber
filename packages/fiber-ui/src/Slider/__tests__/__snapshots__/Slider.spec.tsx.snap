// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
Array [
  <div>
    <div
      className="sc-bdnylx eHCNGe"
      defaultValue={30}
      max={100}
      min={0}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      step={1}
    >
      <div
        className="sc-gtssRu uGDIs"
      />
      <div
        className="sc-dlnjPT hkexCO"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="sc-hKFyIo fToxvM"
        style={
          Object {
            "left": "calc(30% - 7px)",
          }
        }
      />
    </div>
    <br />
    <div
      className="sc-bdnylx eHCNGe"
      defaultValue={0}
      max={100}
      min={0}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      step={1}
      value={30}
    >
      <div
        className="sc-gtssRu uGDIs"
      />
      <div
        className="sc-dlnjPT hkexCO"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="sc-hKFyIo fToxvM"
        style={
          Object {
            "left": "calc(30% - 7px)",
          }
        }
      />
    </div>
  </div>,
  <div>
    <div
      className="sc-bdnylx eHCNGe"
      defaultValue={30}
      max={100}
      min={0}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      step={2}
    >
      <div
        className="sc-gtssRu uGDIs"
      />
      <div
        className="sc-dlnjPT hkexCO"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="sc-hKFyIo fToxvM"
        style={
          Object {
            "left": "calc(30% - 7px)",
          }
        }
      />
    </div>
    <br />
    <div
      className="sc-bdnylx eHCNGe"
      defaultValue={30}
      max={100}
      min={0}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      step={10}
    >
      <div
        className="sc-gtssRu uGDIs"
      />
      <div
        className="sc-dlnjPT hkexCO"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="sc-hKFyIo fToxvM"
        style={
          Object {
            "left": "calc(30% - 7px)",
          }
        }
      />
    </div>
  </div>,
  <div>
    <div
      className="sc-bdnylx jflDPR"
      defaultValue={30}
      disabled={true}
      max={100}
      min={0}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      step={1}
    >
      <div
        className="sc-gtssRu uGDIs"
      />
      <div
        className="sc-dlnjPT hIvuBx"
        disabled={true}
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="sc-hKFyIo iRBwDR"
        disabled={true}
        style={
          Object {
            "left": "calc(30% - 7px)",
          }
        }
      />
    </div>
  </div>,
  <div>
    <div
      style={
        Object {
          "height": 300,
        }
      }
    >
      <div
        className="sc-bdnylx fBsVGD"
        defaultValue={30}
        max={100}
        min={0}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        step={1}
      >
        <div
          className="sc-gtssRu kgRVHX"
        />
        <div
          className="sc-dlnjPT iqcJPQ"
          style={
            Object {
              "height": "30%",
            }
          }
        />
        <div
          className="sc-hKFyIo qiLKQ"
          style={
            Object {
              "top": "calc(70% - 7px)",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div>
    <div
      className="sc-bdnylx EIVIZ"
      defaultValue={30}
      max={100}
      min={0}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      step={1}
    >
      <div
        className="sc-gtssRu uGDIs"
      />
      <div
        className="sc-dlnjPT hkexCO"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="sc-hKFyIo fToxvM"
        style={
          Object {
            "left": "calc(30% - 7px)",
          }
        }
      />
      <span
        className="sc-eCApGN bTQucm"
        style={
          Object {
            "left": "calc(0% + 0px)",
          }
        }
      >
        0°C
      </span>
      <span
        className="sc-eCApGN bTQucm"
        style={
          Object {
            "left": "calc(26% + 0px)",
          }
        }
      >
        26°C
      </span>
      <span
        className="sc-eCApGN bTQucm"
        style={
          Object {
            "left": "calc(37% + 0px)",
          }
        }
      >
        37°C
      </span>
      <span
        className="sc-eCApGN bTQucm"
        style={
          Object {
            "color": "#f50",
            "left": "calc(100% + 0px)",
          }
        }
      >
        <strong>
          100°C
        </strong>
      </span>
    </div>
    <br />
    <div
      style={
        Object {
          "height": 300,
        }
      }
    >
      <div
        className="sc-bdnylx bIjtZM"
        defaultValue={30}
        max={100}
        min={0}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        step={1}
      >
        <div
          className="sc-gtssRu kgRVHX"
        />
        <div
          className="sc-dlnjPT iqcJPQ"
          style={
            Object {
              "height": "30%",
            }
          }
        />
        <div
          className="sc-hKFyIo qiLKQ"
          style={
            Object {
              "top": "calc(70% - 7px)",
            }
          }
        />
        <span
          className="sc-eCApGN lbCWBO"
          style={
            Object {
              "top": "calc(100% - 10px)",
            }
          }
        >
          0°C
        </span>
        <span
          className="sc-eCApGN lbCWBO"
          style={
            Object {
              "top": "calc(74% - 10px)",
            }
          }
        >
          26°C
        </span>
        <span
          className="sc-eCApGN lbCWBO"
          style={
            Object {
              "top": "calc(63% - 10px)",
            }
          }
        >
          37°C
        </span>
        <span
          className="sc-eCApGN lbCWBO"
          style={
            Object {
              "color": "#f50",
              "top": "calc(0% - 10px)",
            }
          }
        >
          <strong>
            100°C
          </strong>
        </span>
      </div>
    </div>
  </div>,
]
`;
